src/oom/oomd-manager.c

```diff
@@ -232,6 +232,25 @@ static int get_monitored_cgroup_contexts_candidates(Hashmap *monitored_cgroups,
         return 0;
 }
 
+static int update_monitored_cgroup_contexts_candidates(Hashmap *monitored_cgroups, Hashmap **candidates) {
+        _cleanup_hashmap_free_ Hashmap *new_candidates = NULL;
+        int r;
+
+        assert(monitored_cgroups);
+        assert(candidates);
+
+        r = get_monitored_cgroup_contexts_candidates(monitored_cgroups, &new_candidates);
+        if (r == -ENOMEM)
+                return r;
```

Comment:
what's with the other error conditions? are they knowinly ignored? if so, add a comment. or better: debug log about this, saying ", ignoring: %m" in the message